ctpolicy/chromepolicy_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ctpolicy
+
+import (
+	"testing"
+
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+func TestCheckChromePolicy(t *testing.T) {
+	var builder СhromePolicyBuilder
+	chrm := builder.Build(&loglist.SampleLogList)
+
+	groups := chrm.Groups()
+	if len(groups) != 3 {
+		t.Errorf("Chrome-policy builder is expected to creat 3 log-groups while it created %d.", len(groups))
+	}
+	for groupname, loggroup := range groups {
+		if groupname == "Base" {
+			if len(loggroup.logURLMap) != len(loglist.SampleLogList.Logs) {
+				t.Errorf("Base group of formed policy is expected to contain %d logs while it contains %d.", len(loglist.SampleLogList.Logs), len(loggroup.logURLMap))
+			}
+		} else if groupname == "Google-operated" {
+			if len(loggroup.logURLMap) != 4 {
+				t.Errorf("Google-group of formed policy is expected to contain 4 logs while it contains %d.", len(loggroup.logURLMap))
+			}
+		} else if groupname == "Non-Google-operated" {
+			if len(loggroup.logURLMap) != 1 {
+				t.Errorf("Non-Google-group of formed policy is expected to contain 1 log while it contains %d.", len(loggroup.logURLMap))
+			}
+		} else {
+			t.Errorf("Unexpected Log-group name: %q", groupname)
+		}
+	}
+	// make some certificate with roots-required of different lifetime.
```

Comment:
Express in the form of a `TODO(Mercurrent)`...

Reply:
Moved out. As lifetime covered in a neighbour test and root-check is out of current scope.